### PR TITLE
C++ demangling on MinGW

### PR DESCRIPTION
**Pull request by BrettDong**
#### Summary
Infrastructure "C++ symbol demangling in crash reports on MinGW"

#### Purpose of change
Expand C++ symbol demangling support to MinGW.

#### Describe the solution
* Extract symbol demangling to a single function `demangle()`.
* Call that function from `_WIN32` code path as well.

#### Testing
Wait and see crash log from MinGW cross compile build in Travis CI.

### CODE REVIEW

File: tests/test_main.cpp [hunk 1]
```diff
@@ -281,6 +281,10 @@ CATCH_REGISTER_LISTENER( CataListener )
 
 int main( int argc, const char *argv[] )
 {
+#ifdef _WIN32
+    int *p = NULL;
+    *p = 0;
+#endif
```
Comment (Qrox) [inline tests/test_main.cpp:287]: The compiler may just optimize this obvious null dereference away (It's undefined behavior, not guaranteed crash). `std::abort` can be used instead to crash the program.

File: src/debug.cpp [hunk 2]
```diff
@@ -837,7 +854,7 @@ void debug_write_backtrace( std::ostream &out )
         out << "\n  #" << i;
         out << "\n    (dbghelp: ";
         if( SymFromAddr( proc, reinterpret_cast<DWORD64>( bt[i] ), &off, &sym ) ) {
-            out << sym.Name << "+0x" << std::hex << off << std::dec;
+            out << demange( sym.Name ) << "+0x" << std::hex << off << std::dec;
```
Comment (Qrox) [inline src/debug.cpp:857]:
```suggestion
            out << demangle( sym.Name ) << "+0x" << std::hex << off << std::dec;
```

**Comment by BrettDong** [comment]
AppVeyor times out.

**Comment by BrettDong** [comment]
Turns out that symbol demangling in this way works in `.exe` natively compiled by GCC on Windows, but isn't functioning in cross-compiled binary.

**Pull request merged**